wireapp/wire-android-reloaded
# fix: keyboard behavior on "remove device" screen and copy on "register device" screen [AR-1478] [AR-1532]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1478" title="AR-1478" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />AR-1478</a>  Remove Device - Keyboard is open after canceling remove device popup
  </td></table>
  <br />
 














----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

The issue with keyboard being still open after dismissing the "remove device" dialog kept occuring after a previous attempt to fix it. The code itself wasn't pretty so there was a space for the refactor.
Also, the copy on "register device" was updated on the designs.

### Causes (Optional)

It looks like dialogs have its own `SoftwareKeyboardController` different than the one used in a screen that shows such dialog and in order to close the keyboard that was open by the dialog, it's necessary to use the controller from such dialog so it needs to be executed inside the dialog's content.

### Solutions

After several approaches and ideas ( i.a. using `hideKeyboard` as a state param, using `DisposableEffect`, passing custom `SoftwareKeyboardController` to have more control over it), looks like this one is the only one that works as expected. It stores the controller from dialog's content in its parent to be able to use that when dismissing the dialog and passes the `hideKeyboard` function to ViewModel to be able to execute it when removal succeeded and before it navigates to the next screen. 
The copy on "register device" was updated to match the designs.

### Testing

#### How to Test

1. There must be more than 7 devices registered
2. Try to login
3. See remove device screen
4. Tap on a device to remove
5. Type password
6. Tap Cancel

To get the "register device" screen, after step 3 kill the app and launch it again.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
